rocketpy/motors/Motor.py

```diff
@@ -0,0 +1,1373 @@
+
+        # Return all extract content
+        return comments, description, dataPoints
+
+    def exportEng(self, fileName, motorName):
+        """Exports thrust curve data points and motor description to
+        .eng file format. A description of the format can be found
+        here: http://www.thrustcurve.org/raspformat.shtml
+
+        Parameters
+        ----------
+        fileName : string
+            Name of the .eng file to be exported. E.g. 'test.eng'
+        motorName : string
+            Name given to motor. Will appear in the description of the
+            .eng file. E.g. 'Mandioca'
+
+        Returns
+        -------
+        None
+        """
+        # Open file
+        file = open(fileName, "w")
+
+        # Write first line
+        file.write(
+            motorName
+            + " {:3.1f} {:3.1f} 0 {:2.3} {:2.3} RocketPy\n".format(
+                2000 * self.grainOuterRadius,
+                1000
+                * self.grainNumber
+                * (self.grainInitialHeight + self.grainSeparation),
+                self.propellantInitialMass,
+                self.propellantInitialMass,
+            )
+        )
+
+        # Write thrust curve data points
+        for time, thrust in self.thrust.source[1:-1, :]:
+            # time, thrust = item
+            file.write("{:.4f} {:.3f}\n".format(time, thrust))
+
+        # Write last line
+        file.write("{:.4f} {:.3f}\n".format(self.thrust.source[-1, 0], 0))
+
+        # Close file
+        file.close()
+
+        return None
+
+    def info(self):
+        """Prints out a summary of the data and graphs available about the
+        Motor.
+
+        Parameters
+        ----------
+        None
+
+        Return
+        ------
+        None
+        """
+        # Print motor details
+        print("\nMotor Details")
+        print("Total Burning Time: " + str(self.burnDuration) + " s")
+        print(
+            "Total Propellant Mass: "
+            + "{:.3f}".format(self.propellantInitialMass)
+            + " kg"
+        )
+        print(
+            "Propellant Exhaust Velocity: "
+            + "{:.3f}".format(self.exhaustVelocity.average(*self.burn_time))
+            + " m/s"
+        )
+        print("Average Thrust: " + "{:.3f}".format(self.averageThrust) + " N")
+        print(
+            "Maximum Thrust: "
+            + str(self.maxThrust)
+            + " N at "
+            + str(self.maxThrustTime)
+            + " s after ignition."
+        )
+        print("Total Impulse: " + "{:.3f}".format(self.totalImpulse) + " Ns")
+
+        # Show plots
+        print("\nPlots")
+        self.thrust()
+
+        return None
+
+    @abstractmethod
+    def allInfo(self):
+        """Prints out all data and graphs available about the Motor.
+
+        Parameters
+        ----------
+        None
+
+        Return
+        ------
+        None
+        """
+        pass
+
+
+class GenericMotor(Motor):
```

Comment:
Add docstring here because I honestly don't have any idea why this thing exists.

Reply:
Added on 33c46c8b333d3fbde74423088d4f761000d66e69.